kernel/filesystem/FileCache.cpp

```diff
@@ -3,38 +3,45 @@
 #include <arch/Platform.h>
 #include <debug/Log.h>
 #include <memory/Pmm.h>
+#include <memory/virtual/VfsVmDriver.h>
 #include <UnitConverter.h>
+#include <Memory.h>
 #include <Maths.h>
 
 namespace Npk::Filesystem
 {
-    size_t cacheUnitSize;
-    
+    FileCacheInfo cacheInfo;
+
     void InitFileCache()
     {
         //find the smallest granularity supported by the MMU, we'll use that as a cache unit size.
-        cacheUnitSize = -1ul;
+        cacheInfo.unitSize = -1ul;
         const HatLimits& mmuLimits = GetHatLimits();
         for (size_t i = 0; i < mmuLimits.modeCount; i++)
         {
-            if (mmuLimits.modes[i].granularity < cacheUnitSize)
-                cacheUnitSize = mmuLimits.modes[i].granularity;
+            if (mmuLimits.modes[i].granularity < cacheInfo.unitSize)
```

Comment:
this is probably a facepalm for me but -1ul what does it mean? 

I suppose is -1 ? 

If yes are there cases where the if is ever true?

Reply:
This code finds the smallest mode (read: page size) supported by the mmu. The modes aren't gaureteneed to be in order so it's just looping through them until it finds the one it wants.
The initial value or -1 wraps around to mean all bits set/Max value for that variable. It's simplifies the code in the loop a bit, rather than also checking if the unit size is valid, we just check if the current mode is smaller than it.

Reply:
If unit size is never changed then we have a problem in the mmu code and the kernel probably shouldn't be running :p